web/src/peer_data.ts

```diff
@@ -300,7 +332,10 @@ export function bulk_remove_subscribers({
     for (const stream_id of stream_ids) {
         const subscribers = get_loaded_subscriber_subset(stream_id);
         for (const user_id of user_ids) {
-            subscribers.delete(user_id);
+            if (subscribers.has(user_id)) {
+                subscribers.delete(user_id);
+                sub_store.get(stream_id)!.subscriber_count -= 1;
+            }
```

Comment:
I'm a bit skeptical of this -- most of the users who are subscribers will not be in our local matrix. So it seems like it will drift the `subscriber_count` whenever a user we don't have local subscriber data for gets removed.

I think probably we want to do `sub_store.get(stream_id)!.subscriber_count -= 1;` unconditionally if we don't have fully loaded data for that channel.

Reply:
Ah yeah, I agree. Thanks!